client/packages/inventory/src/Stocktake/ListView/NewCreateStocktakeModal.tsx

```diff
@@ -0,0 +1,343 @@
+  const [
+    { location, masterList, vvmStatus, expiryDate, itemStatus },
+    setState,
+  ] = useState<ModalState>({
+    location: null,
+    vvmStatus: null,
+    masterList: null,
+    expiryDate: null,
+    itemStatus: ItemStatus.InStock,
+  });
+
+  const stockFilter: StockLineFilterInput = {
+    location: location && {
+      id: { equalTo: location.id },
+    },
+    masterList: masterList && {
+      id: { equalTo: masterList.id },
+    },
+    expiryDate: expiryDate && {
+      beforeOrEqualTo: Formatter.naiveDate(expiryDate),
+    },
+    vvmStatusId: vvmStatus && {
+      equalTo: vvmStatus.id,
+    },
+    hasPacksInStore: true,
+  };
+
+  const { data } = useStockListCount(stockFilter);
+  const { data: masterListLineCount } = useMasterListLineCount(masterList?.id);
+
+  const { localisedDate } = useFormatDateTime();
+
+  const generateComment = () => {
+    if (itemStatus === ItemStatus.None) return '';
+
+    const filterComments: string[] = [];
+
+    if (!!masterList) {
+      filterComments.push(
+        t('stocktake.master-list-template', {
+          masterList: masterList.name,
+        })
+      );
+    }
+    if (!!location) {
+      filterComments.push(
+        t('stocktake.location-template', {
+          location: location.code,
+        })
+      );
+    }
+    if (!!expiryDate) {
+      filterComments.push(
+        t('stocktake.expires-before-template', {
+          date: localisedDate(expiryDate),
+        })
+      );
+    }
+    if (!!vvmStatus) {
+      filterComments.push(
+        t('stocktake.vvm-status-template', {
+          vvmStatus: vvmStatus.description,
+        })
+      );
+    }
+
+    if (filterComments.length === 0) return undefined;
+    if (filterComments.length === 1)
+      return t('stocktake.comment-template', { filters: filterComments[0] });
+
+    const comments = t('stocktake.comment-and-template', {
+      start: filterComments.slice(0, -1).join(', '),
+      end: filterComments[filterComments.length - 1],
+    });
+
+    return t('stocktake.comment-template', { filters: comments });
+  };
+
+  const onSave = () => {
+    // Our API only has a `beforeOrEqualTo` filter, so just kludging the date back 1 day here
+    const adjustedExpiryDate = expiryDate
+      ? DateUtils.addDays(expiryDate, -1)
+      : null;
+
+    const args: CreateStocktakeInput = {
+      masterListId: masterList?.id,
+      locationId: location?.id,
+      vvmStatusId: vvmStatus?.id,
+      expiresBefore: Formatter.naiveDate(adjustedExpiryDate),
+      isInitialStocktake: false,
+      description,
+      comment: generateComment(),
+    };
+    onCreate(args).then(id => {
+      if (id) {
+        navigate(id);
+      }
+    });
+  };
+
+  const estimateLineCount = (suggestedStatus?: ItemStatus): number => {
+    const stockCount = data?.totalCount ?? 0;
+    if (suggestedStatus === ItemStatus.All) {
+      return masterListLineCount
+        ? Math.max(masterListLineCount, stockCount)
+        : stockCount;
```

Comment:
I don't think this was your changes, but would be good to fix...

When I select a master list, `All` is the number of items in the master list:
<img width="713" height="613" alt="Screenshot 2025-09-04 at 1 57 33 PM" src="https://github.com/user-attachments/assets/6cd5824d-5263-4f5b-8630-6e906f3a09bb" />

But when no master list selected, all only shows the number of lines I have in stock... I think it should show the total number of items visible in my store?
<img width="671" height="586" alt="Screenshot 2025-09-04 at 1 57 41 PM" src="https://github.com/user-attachments/assets/e8c04e3b-b121-4326-8d69-90cfbc59f9bb" />